proto/cluster.proto

```diff
@@ -0,0 +1,74 @@
+syntax = "proto3";
+
+option go_package = "github.com/ray-project/kuberay/proto/go_client";
+package proto;
+
+import "google/protobuf/timestamp.proto";
+
+
+message Cluster {
+  // Required input field. Unique Cluster name provided by user.
+  string name = 1;
+
+  // Required input field. Cluster's namespace provided by user
+  string namespace = 2;
+
+  // Required field. This field indicates the user who owns the cluster.
+  string user = 3;
+
+  // Optional input field. Ray cluster version
+  string version = 4;
+
+  // Optional field.
+  enum Environment {
+    DEV = 0;
+    TESTING = 1;
+    STAGING = 2;
+    PRODUCTION = 3;
+  }
+  Environment environment = 5;
+  
+  // Required field. This field indicates ray cluster configuration
+  ClusterSpec cluster_spec = 6;
+
+  // Output. The time that the Cluster created.
+  google.protobuf.Timestamp created_at = 7;
+
+  // Output. The time that the Cluster deleted.
+  google.protobuf.Timestamp deleted_at = 8;
+}
+
+message ClusterSpec {
```

Comment:
There're only `HeadGroupSpec` and `WorkerGroupSpec` object under `ClusterSpec`, do you think this is helpful? We can also  move `HeadGroupSpec` and `WorkerGroupSpec` directly to `message Cluster`. I am ok with both way @chenk008

Reply:
Yes ,we can simplify it to move `HeadGroupSpec` and `WorkerGroupSpec` into `Cluster`.